crates/turbo-tasks-fs/src/rope.rs

```diff
@@ -532,31 +538,85 @@ impl DeterministicHash for RopeElem {
     }
 }
 
-/// Implements the Read/AsyncRead/Stream/Iterator trait over a Rope.
+/// Implements the [Read]/[AsyncRead]/[Iterator] trait over a [Rope].
 #[derive(Debug, Default)]
-pub struct RopeReader {
+pub struct RopeReader<'a> {
     /// The Rope's tree is kept as a cloned stack, allowing us to accomplish
     /// incremental yielding.
-    stack: Vec<StackElem>,
+    stack: Vec<StackElem<'a>>,
+
+    /// The leaked root allows us to invert the ownership of Bytes. By leaking
+    /// the Rope's root, we're able to use a 'static lifetime and let the
+    /// RopeReader own the memory and cleanup. When the RopeReader is dropped,
+    /// the InnerRope's Arc will be manually decremented to free the memory.
+    leaked_root: Option<*mut InnerRope>,
```

Comment:
I'm not sure I understand the `Box::leak`/`Box::from_raw` dance. Wouldn't it be enough to keep an `Option<InnerRope>` here?

Reply:
Yes, but we still need to turn the `'a` lifetime into a `'static`. Actually, I think this'll work with a `transmute()` and the held `InnerRope`, let me try…